modules/data.land/R/DataONE_doi_download.R

```diff
@@ -0,0 +1,119 @@
+#' @param CNode default is "PROD"
+#' 
+#' @return return the resource_map or a message indicating that there is no corresponding resource_map for the given id
+#' @export
+#'
+#' @examples
+get.resource.map = function(id, CNode = "PROD"){
+  cn <- dataone::CNode(CNode) 
+  locations <- dataone::resolve(cn, pid = id) 
+  mnId <<- locations$data[1,"nodeIdentifier"] # store mnId in global environment
+  mn <<- dataone::getMNode(cn, mnId) # store mn in global environment
+  
+  format.identifier(id) # format the identifier in solr Query format
+  queryParamList <- list(q=doi1, fl="resourceMap") # custom query for the resourceMap
+  resource_map_df <- dataone::query(cn, solrQuery = queryParamList, as="data.frame") 
+  resource_map <<- resource_map_df[1,1] # store resource map in global env. resource map is always in resource_map_df[1,1]
+  
+  if (is.null(resource_map_df[1,1])){ # inform user if id/ doi has a corresponding resource_map or if this needs to be found manually
+    print("doi does not resolve a resource_map. Please manually search for the resource_map in DataONE search: https://search.DataONE.org/#data")
+  } else{
+    print("Continue to next phase to complete download")
+    return(resource_map)
+  }
+} # end function
+
+# -----------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+#' download.packages
+#'
+#' @param resource_map the resource map that corresponds to the given data package
+#' @param CNode defaults to "PROD"
+#' @param download_format typically "application/bagit-097". Other possible formats currently unknown.
+#' @param overwrite_directory boolean that indicates whether or not the function should overwrite the directory
+#' @param directory location that download.packages places the data
+#'
+#' @return results of download
+#' @export
+#'
+#' @examples
+download.package.rm = function(resource_map, directory, CNode = "PROD", download_format = "application/bagit-097", 
+                               overwrite_directory = TRUE){
+  # Finding the mnId (query)
+  cn <- dataone::CNode(CNode) 
+  locations <- dataone::resolve(cn, pid = resource_map) 
+  mnId <<- locations$data[1,"nodeIdentifier"]
+  
+  # download the bagitFile 
+  mn <<- dataone::getMNode(cn, mnId)
+  bagitFile <<- dataone::getPackage(mn, id = resource_map, format = download_format)
+  bagitFile
+  
+
+  zip_contents <<- utils::unzip(bagitFile, files = NULL, list = TRUE, overwrite = TRUE, # list files in bagitFile
+                         junkpaths = FALSE, exdir = "downloads", unzip = "internal",
+                         setTimes = FALSE)
+  
+  utils::unzip(bagitFile, files = NULL, list = FALSE, overwrite = overwrite_directory, # Unzip the bagitFile and store in directory specified under exdir
+        junkpaths = FALSE, exdir = directory, unzip = "internal",
+        setTimes = FALSE)
+  return(zip_contents)
```

Comment:
This relates to what we talked about yesterday - where the ziped folder may contain multiple files but we need to figure out which of the files is the actual data and what format that data is actually in.